tree/ntupleutil/v7/src/RNTupleImporterCLI.cxx

```diff
@@ -0,0 +1,197 @@
+         argState = EArgState::kTreePath;
+      } else if (arg == "--outfile" || arg == "-o") {
+         argState = EArgState::kNTuplePath;
+      } else if (arg == "--rntuple" || arg == "-r") {
+         argState = EArgState::kNTupleName;
+      } else if (arg == "--compression" || arg == "-c") {
+         argState = EArgState::kCompression;
+      } else if (arg == "--unzipped-page-size") {
+         argState = EArgState::kUnzippedPageSize;
+      } else if (arg == "--zipped-cluster-size") {
+         argState = EArgState::kZippedClusterSize;
+      } else if (arg == "--max-unzipped-cluster-size") {
+         argState = EArgState::kUnzippedClusterSize;
+      } else if (arg == "--convert-dots") {
+         config.fConvertDots = true;
+      } else if (arg == "--verbose" || arg == "-v") {
+         config.fVerbose = true;
+      } else if (arg[0] == '-') {
+         std::cerr << "Unrecognized option '" << arg << "'\n";
+         return {};
+      } else {
+         switch (argState) {
+         case EArgState::kTreeName: config.fTreeName = arg; break;
+         case EArgState::kTreePath: config.fTreePath = arg; break;
+         case EArgState::kNTuplePath: config.fNTuplePath = arg; break;
+         case EArgState::kNTupleName: config.fNTupleName = arg; break;
+         case EArgState::kCompression: config.fNTupleOpts.SetCompression(std::stoi(arg)); break;
+         case EArgState::kUnzippedPageSize: config.fNTupleOpts.SetApproxUnzippedPageSize(std::stol(arg)); break;
+         case EArgState::kZippedClusterSize: config.fNTupleOpts.SetApproxZippedClusterSize(std::stol(arg)); break;
+         case EArgState::kUnzippedClusterSize: config.fNTupleOpts.SetMaxUnzippedClusterSize(std::stol(arg)); break;
+         default: std::cerr << "Unrecognized option '" << arg << "'\n"; return {};
+         }
+      }
+   }
+
+   if (config.fTreeName == "") {
+      std::cerr << "Please provide the name of the TTree to convert.\n\n" << usageText << "\n";
+      return {};
+   } else if (config.fTreePath == "") {
+      std::cerr << "Please provide the name of the ROOT file containing the TTree to convert.\n\n" << usageText << "\n";
+      return {};
+   } else if (config.fNTuplePath == "") {
+      std::cerr << "Please provide the name of the ROOT file to write the converted RNTuple to.\n\n"
+                << usageText << "\n";
+      return {};
+   }
+
+   config.fShouldRun = true;
+   return config;
+}
+
+ImporterConfig ROOT::Experimental::RNTupleImporterCLI::ParseArgs(int argc, char **argv)
+{
+   std::vector<std::string> args;
+   args.reserve(argc);
+
+   for (int i = 0; i < argc; ++i) {
+      args.emplace_back(argv[i]);
+   }
+
+   return ParseArgs(args);
+}
+
+void ROOT::Experimental::RNTupleImporterCLI::RunImporter(const ImporterConfig &config)
+{
+#ifdef R__USE_IMT
+   ROOT::EnableImplicitMT();
+#endif
+
+   auto importer = RNTupleImporter::Create(config.fTreePath, config.fTreeName, config.fNTuplePath);
```

Comment:
I think here we should use the `Create` method that takes a `TFile`, and open/create the `TFile` before. With the current factory method, we cannot append an RNTuple to an existing file.

Reply:
There's no `Create` method that takes a `TFile` directly, only one that takes a `TTree`. I added an extra test that appends to an existing file and that appears to work fine.